simulation/simple_sensor_simulator/src/sensor_simulation/detection_sensor/detection_sensor.cpp

```diff
@@ -127,9 +129,16 @@ void DetectionSensor<autoware_auto_perception_msgs::msg::DetectedObjects>::updat
           }
         }
         if (not is_ego) {
-          simulation_interface::toMsg(s.bounding_box().dimensions(), object.shape.dimensions);
           geometry_msgs::msg::Pose pose;
           simulation_interface::toMsg(s.pose(), pose);
+          auto rotation = quaternion_operation::getRotationMatrix(pose.orientation);
+          geometry_msgs::msg::Point center_point;
+          simulation_interface::toMsg(s.bounding_box().center(), center_point);
+          Eigen::Vector3d center(center_point.x, center_point.y, center_point.z);
+          center = rotation * center;
+          pose.position.x = pose.position.x + center.x();
+          pose.position.y = pose.position.y + center.y();
+          pose.position.z = pose.position.z + center.z();
```

Comment:
These lines are copied from here, so it may be better to define new function for coordinate conversion of bounding box.

https://github.com/tier4/scenario_simulator_v2/blob/5814fe79978efb64b85f166083dad0f69d783c93/simulation/simple_sensor_simulator/src/sensor_simulation/occupancy_grid/occupancy_grid_sensor.cpp#L104-L113